src/MiniProfiler/ProfileProviders/WebRequestProfilerProvider.cs

```diff
@@ -150,7 +150,23 @@ private static void EnsureName(MiniProfiler profiler, HttpRequest request)
         /// </summary>
         public override MiniProfiler GetCurrentProfiler() => Current;
 
+        /// <summary>
+        /// Current head timing.
+        /// </summary>
+        public override Timing CurrentHead
```

Comment:
I think maybe the behavior change concern can be mitigated by introducing AsyncWebRequestProfilerProvider that would support the async operations while WebRequestProfilerProvider can stay unchanged. As a result the switch to the new behavior would be an explicit action required from the user of the library